### DISCUSSION TITLE
Engelmann Sensostar Q -> issues getting all Fields

### DISCUSSION THREAD

**Discussion by mschmiedel** [arendst/Tasmota#22022] [Q&A] [answered]
I got myself a Wifi opti reader for reading out the Sensostar Q. I used the Sensostar E script, which seems to be somehow working. Issue is that I only get 4 valid values: supply temp, return temp, temp difference, and days in service. The interesting data (energy, flow, etc.) are all on 0. 

I am guessing that I need to do some adjustments to the mapping of the fields, but since I am fairly new to playing around with this kind of interfaces, I am not sure on what to do with the binary output that is being dumped by the script. 

Can someone point me to some resources on how to get this reverse engineered? I was guessing that the hex data was a mbus data packet, but it does not seem so. :)

**Comment by Jason2866** [comment]
About what are you talking at all? No link, no description.

**Reply by mschmiedel** [reply]
I am using the Script for the Sensostar E from here:

https://tasmota.github.io/docs/Smart-Meter-Interface/#metcom-mcs301 (no direct link to the Sensostar E)

The device I want to read out is this one: https://www.engelmann.de/wp-content/uploads/2024/03/Datenblatt_SensoStar_Q_2023_12_01_de.pdf

I am using the Hichi WiFi, and since I get some data with correct readings (see above), the overall communication seems to be working...

I had assumed they use the same interface, but there seems to be a difference as I am missing some of the fields... ;) 

So I am guessing that there must be a difference with the mapping. What I am asking is for some insight on how to reverse engineer the read out data from the meter to find the missing values...

**Reply by sfromis** [reply]
I was wondering if you got the script the right way. The way it was shown in the docs page you linked was garbled, due to bad formatting in the page source. I've now fixed that formatting issue, so that what you see on the docs page looks better now (you may have to refresh the page). I'd suggest that you ensure that the script you have matches the better formatting on the docs page, or just copy it over again.

**Reply by mschmiedel** [reply]
> I was wondering if you got the script the right way. The way it was shown in the docs page you linked was garbled, due to bad formatting in the page source. I've now fixed that formatting issue, so that what you see on the docs page looks better now (you may have to refresh the page). I'd suggest that you ensure that the script you have matches the better formatting on the docs page, or just copy it over again.

I rechecked, but I guess the script I got is fine, I do not see any formatting issues, what I have (and had) seems to cover the specs. I just guess that the device I have has some different field mappings. However, since I have no experience with this kind of serial communication and mbus, I am struggling to get the proper data packets and reverse engineer it, this is where I am hoping for some push(es) in the right direction. :)

**Reply by mschmiedel** [reply, marked as answer]
OK, I got it to work. At the end, I installed a newer Firmware with Tasmota 13.1.0, with this one I get correct readings directly, also the dataset is being logged every time and seems much more coherent than the one I was getting before. The previous Firmware was 12.3.1...

What I can confirm now is that the Sensostar Q works with the same Script as the Sensostar E, which is coherent with the documentation from Engelsmann...
